### PR TITLE
[test]added unit test cases for network-layer filter "direct_response"

### PR DESCRIPTION
**Pull request by jianwen612**
Additional Description:
[see https://github.com/envoyproxy/envoy/issues/11998]
Added unit test cases for "direct_response".
Removed "direct_response" from test/per_file_coverage.sh

Risk Level:low

Testing: 
covered onData() and onNewConnection() method of the filter.
Coverage for this filter is now 100%

[Optional Fixes #Issue] https://github.com/envoyproxy/envoy/issues/11998

/cc @asraa 
/cc @samkerner

### CODE REVIEW

File: test/extensions/filters/network/direct_response/direct_response_test.cc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  EXPECT_CALL(read_filter_callbacks_.connection_, close(Network::ConnectionCloseType::FlushWrite));
+  EXPECT_CALL(read_filter_callbacks_.connection_.stream_info_,
+              setResponseCodeDetails(StreamInfo::ResponseCodeDetails::get().DirectResponse));
+  EXPECT_EQ(Network::FilterStatus::StopIteration, filter_->onNewConnection());
+}
+
+// Test the filter's onNewConnection() with an empty response
+TEST_F(DirectResponseFilterTest, OnNewConnectionEmptyResponse) {
+  initialize("");
+  EXPECT_CALL(read_filter_callbacks_.connection_, write(testing::_, testing::_)).Times(0);
```
Comment (zuercher) [inline test/extensions/filters/network/direct_response/direct_response_test.cc:42]: You can add `using testing::_` near the top and just use `_` here, if you'd like.

File: test/extensions/filters/network/direct_response/direct_response_test.cc [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+              setResponseCodeDetails(StreamInfo::ResponseCodeDetails::get().DirectResponse));
+  EXPECT_EQ(Network::FilterStatus::StopIteration, filter_->onNewConnection());
+}
+
+// Test the filter's onData()
+TEST_F(DirectResponseFilterTest, OnData) {
+  initialize("hello");
+  Buffer::OwnedImpl data("data");
+  EXPECT_EQ(Network::FilterStatus::Continue, filter_->onData(data, false));
+}
```
Comment (zuercher) [inline test/extensions/filters/network/direct_response/direct_response_test.cc:54]: Should there be an expectation, similar to the one in OnNewConnectionEmptyResponse, that nothing is written to the downstream connection?

Comment (jianwen612) [reply test/extensions/filters/network/direct_response/direct_response_test.cc:56]: > Should there be an expectation, similar to the one in OnNewConnectionEmptyResponse, that nothing is written to the downstream connection?

Thanks! Added the expectation.

File: test/extensions/filters/network/direct_response/direct_response_test.cc (hunk 1 above)
Comment (jianwen612) [reply test/extensions/filters/network/direct_response/direct_response_test.cc]: > You can add `using testing::_` near the top and just use `_` here, if you'd like.

Thanks!

**Comment by zuercher** [comment]
@jianwen612 Apparently azure pipelines won't run tests if there's a merge conflict, so you'll have to merge master to get the tests passing.

**Comment by jianwen612** [comment]
> @jianwen612 Apparently azure pipelines won't run tests if there's a merge conflict, so you'll have to merge master to get the tests passing.



> @jianwen612 Apparently azure pipelines won't run tests if there's a merge conflict, so you'll have to merge master to get the tests passing.



> @jianwen612 Apparently azure pipelines won't run tests if there's a merge conflict, so you'll have to merge master to get the tests passing.

Merged. Thanks!

**Pull request merged**